euphoria-spark/src/test/java/cz/seznam/euphoria/spark/BroadcastHashJoinTest.java

```diff
@@ -0,0 +1,104 @@
+package cz.seznam.euphoria.spark;
+
+import cz.seznam.euphoria.core.client.dataset.Dataset;
+import cz.seznam.euphoria.core.client.io.Collector;
+import cz.seznam.euphoria.core.client.operator.LeftJoin;
+import cz.seznam.euphoria.core.client.operator.RightJoin;
+import cz.seznam.euphoria.core.client.util.Pair;
+import cz.seznam.euphoria.operator.test.JoinTest;
+import cz.seznam.euphoria.operator.test.junit.AbstractOperatorTest;
+import cz.seznam.euphoria.operator.test.junit.ExecutorProviderRunner;
+import cz.seznam.euphoria.operator.test.junit.Processing;
+import cz.seznam.euphoria.shadow.com.google.common.collect.Sets;
+import cz.seznam.euphoria.spark.testkit.SparkExecutorProvider;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+@RunWith(ExecutorProviderRunner.class)
+public class BroadcastHashJoinTest extends AbstractOperatorTest implements SparkExecutorProvider {
```

Comment:
Strongly suggest moving this class to operator test kit and run against all executors. It *must* pass on all of them, regardless if they support, or don't support the hint. It would be good to actually run the same test twice - with hint and without hint and expect the same results.

Reply:
Although it looks like basically the same test is already there. Maybe we can just add another one with the hint. Maybe we could parametrize the join test somehow. I'm not sure how, although. :)